### ISSUE TITLE
Instrument alerting for Cluster nodes disk usage and Cluster memory usage

### ISSUE THREAD

**Issue by dchoi27** [nftstorage/nft.storage#1452]
Sister issue to https://github.com/web3-storage/web3.storage/issues/1029

**Comment by dchoi27** [comment]
@alanshaw @vasco-santos @hugomrdias also might have good thoughts here. There was an incident where our cluster nodes' memory were shooting up through the roof, and the NetOps team was alerted, but we were not. They're ultimately the ones who have to solve it but we should have visibility as well since it affects our service. Might be worth checking in with them on what their thresholds are.

**Comment by dashcraft** [comment]
We should likely move metrics to prom/graphana or house them all internally + timestamps, so we can reduce the massive counting we do across the data points.

tldr move metrics to 1 place, either nft or prom/graf

**Comment by olizilla** [comment]
The alerts are configured in alertmanager which is bundled with prometheus and run on PL k8s infra. The thresholds are defined in this file https://github.com/protocol/monitoring-infra/blob/master/ansible/roles/prometheus-operator/files/alert-rules.yml

```yaml
  - name: instances
    rules:
    - alert: node_high_memory_usage_95_percent
      expr: node_memory_MemAvailable_bytes  / on(instance) (sum(node_memory_MemTotal_bytes) by(instance)) < 0.05
      for: 5m
      annotations:
        description: '{{ $labels.instance }} available memory is {{ humanizePercentage $value }}.'
        summary: Free memory alert for instance '{{ $labels.instance }}'

    - alert: node_filesystem_fill_rate_1h
      # alert if less than 1GB (1000000000 bytes) will be available in 3hr time at current fill rate
      expr: predict_linear(node_filesystem_avail_bytes{mountpoint=~"/|/data"}[1h], 1 * 3600) < 1000000000
      for: 1h
      labels:
        severity: critical
      annotations:
        description: instance '{{ $labels.instance }}' filesystem '{{ $labels.family }}' is going to fill up in 1h.
        summary: Disk fill alert for instance '{{ $labels.instance }}'

    - alert: node_filesystem_inode_fill_rate_5h
      # inodes is 'files' in prometheus (and statfs naming) #https://github.com/prometheus/node_exporter/issues/1392
      expr: predict_linear(node_filesystem_files_free{mountpoint=~"/|/data"}[1h], 5 * 3600) < 10000
      for: 1h
      labels:
        severity: critical
      annotations:
        description: 'instance {{ $labels.instance }} data filesystem is going to run out of inodes in 5h.'
        summary: Disk inodes fill alert for instance '{{ $labels.instance }}'

    - alert: node_filesystem_full
      # alert if less than 7% of disk space is available. FYU ext4 reserves 5% of disk space for the root user
      expr: node_filesystem_free_bytes / node_filesystem_size_bytes < 0.07
      for: 15m
      labels:
        severity: critical
      annotations:
        description: 'instance {{ $labels.instance }} filesystem {{ $labels.mountpoint }} is very low on space: {{ humanizePercentage $value }}'
        summary: Disk fill alert for instance '{{ $labels.instance }}'

    # Alert threshold for all non-preload nodes
    - alert: node_load15_overload
      # load15 / CPU count
      expr: node_load15{job!~"preload.*"} / on(instance) (count by(instance) (node_cpu_seconds_total{job!~"preload.*", mode="idle"})) > 1.1
      for: 15m
      labels:
        severity: critical
      annotations:
        description: 'Instance {{ $labels.instance }} is under heavy load (15m).'
        summary: 'Excessive load alert {{ $labels.instance }}'

    - alert: process_restart
      expr: changes(process_start_time_seconds[1m]) > 0
      labels:
        severity: critical
      annotations:
        description: 'Process restart on job {{ $labels.job }}: {{ $labels.instance }}'
        summary: 'Process restart on job {{ $labels.job }}: {{ $labels.instance }}'
```

We can set up a slack webhook, and update the the alertmanager-config.yaml with an additional alert reciever block as per https://github.com/protocol/monitoring-infra/blob/master/ansible/roles/prometheus-operator/templates/alertmanager-config.yml#L23-L29

...and add a filter so that only alerts on *.storage hosts are sent there. Doing it this way would re-use the existing system resource level `node_exporter` alert plumbing that exists, so we'd see the same alerts that infra sees, just in a different slack channel.

Alternatively we can create our own similar set of graphs with alerts on in grafana. This might be desirable as we'd see a graph on a dashboard with a threshold on it that was being breached, and the alerts could go to ops-genie as needed.

The prevailing wisdom of PL infra has been to use the alert-manager that is built in to prometheus so that the data collection sink and the alerting are tightly coupled... if we use grafana for alerting then we are relying on prometheus **and** grafana to be functioning correctly to get our alerts in a timely fashion. 

I already duplicated our disk space alerts to grafana on the alerts dash here https://protocollabs.grafana.net/d/cs6uG7Snk/web3-storage-alerts?orgId=1

**Comment by fabiolrodriguez** [comment]
Alerts instrumented on monitoring repo: https://github.com/protocol/monitoring-infra/pull/303

**Comment by fabiolrodriguez** [comment]
Alerts up and running:

![image](https://user-images.githubusercontent.com/15051788/168673437-2b6fe537-fd59-40a0-be82-023f3a3c7314.png)
